src/pages/sales-settings/Form/SalesSettingsForm.js

```diff
@@ -0,0 +1,286 @@
+import { Grid } from '@mui/material'
+import * as yup from 'yup'
+import { useContext, useEffect } from 'react'
+import { FormControlLabel, Checkbox } from '@mui/material'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { ControlContext } from 'src/providers/ControlContext'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { DataSets } from 'src/resources/DataSets'
+
+export default function SalesSettingsForm({ _labels, access }) {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const arrayAllow = [
+    'plId',
+    'ptId',
+    'currencyId',
+    'siteId',
+    'maxReturnDays',
+    'SAFIIntegration',
+    'sdpClientName',
+    'sdpItemName',
+    'sdpUnitPrice',
+    'allowSalesNoLinesTrx',
+    'salesTD'
+  ]
+
+  const { formik } = useForm({
+    initialValues: {
+      plId: null,
+      ptId: null,
+      currencyId: null,
+      siteId: null,
+      maxReturnDays: null,
+      SAFIIntegration: null,
+      sdpClientName: false,
+      sdpItemName: false,
+      sdpUnitPrice: false,
+      allowSalesNoLinesTrx: false,
+      salesTD: false
+    },
+    enableReinitialize: false,
+    maxAccess: access,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      maxReturnDays: yup.number().min(0, 'min').max(9999, 'max')
+    }),
+    onSubmit: async obj => {
+      try {
+        var data = []
+
+        Object.entries(obj).forEach(([key, value]) => {
+          if (arrayAllow.includes(key)) {
+            let processedValue = value;
+    
+            if (Array.isArray(value)) {
+              processedValue = value.includes('on');
+            }
+    
+            const newObj = { key: key, value: processedValue };
+            data.push(newObj);
+          }
+        });
+
+        const response = await postRequest({
+          extension: SystemRepository.Defaults.set,
+          record: JSON.stringify({ sysDefaults: data })
+        })
+
+        if (response) {
+          toast.success(platformLabels.Added)
```

Comment:
shouldn't this be edited instead of added?

Reply:
Yes